lib/inspec/secrets/yaml.rb

```
@@ -25,8 +25,8 @@ def initialize(target)
       end
 
       if @inputs == false || !@inputs.is_a?(Hash)
-        Inspec::Log.warn("#{self.class} unable to parse #{target}: invalid YAML or contents is not a Hash")
         @inputs = nil
+        raise("#{self.class} unable to parse #{target}: invalid YAML or contents is not a Hash")
```

We should raise the error if the yaml is corrupted or it's empty.

---

Does this help address CHEF-3849 ? - abort immediately if waiver file is corrupt?

---

Yes.